pkg/cmd/image/save.go

```diff
@@ -0,0 +1,90 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package image
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/images/archive"
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/idutil/imagewalker"
+	"github.com/containerd/nerdctl/pkg/platformutil"
+	"github.com/containerd/nerdctl/pkg/strutil"
+	"github.com/mattn/go-isatty"
+)
+
+func Save(ctx context.Context, client *containerd.Client, images []string, options types.ImageSaveOptions) error {
+	writer := options.Stdout
+	if options.Output != "" {
+		f, err := os.OpenFile(options.Output, os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		writer = f
```

Comment:
should we update the `options.Stdout` in place instead of passing a `wirter` to `saveImages`? Otherwise, `saveImages` has two `io.writer` inputs (`writer` and `opt.Stdout`) and sometimes they're the same and sometimes not.

Reply:
Make sense. But their types are not always the same.